tests/unit/test_config.py

```diff
@@ -41,6 +41,14 @@ def test_malformed_yaml() -> None:
     )
 
 
+def test_missing_config_file() -> None:
+    """Check how missing configuration file is handled."""
+    with pytest.raises(Exception, match="Not a directory"):
+        # /dev/null is special file so it can't be directory
+        # at the same moment
+        config.init_config("/dev/null/non-existent")
```

Comment:
@tisnik config.init_config() doesn't propagate Exception https://github.com/openshift/lightspeed-service/blob/9863189cf15291b2ea2f9be76dc21fc8d83e3418/ols/utils/config.py#L56
I'm not sure how this test passes for you. When I try to run it locally I'm seeing 
```
_________________________________________________________ test_missing_config_file _________________________________________________________

    def test_missing_config_file() -> None:
        """Check how missing configuration file is handled."""
>       with pytest.raises(Exception, match="Not a directory"):
E       Failed: DID NOT RAISE <class 'Exception'>

tests/unit/test_config.py:46: Failed
----------------------------------------------------------- Captured stdout call -----------------------------------------------------------
Failed to load config file /dev/null/non-existent: [Errno 20] Not a directory: '/dev/null/non-existent'
Traceback (most recent call last):
  File "/home/syedriko/lightspeed-service/ols/utils/config.py", line 48, in init_config
    with open(config_file, "r") as f:
         ^^^^^^^^^^^^^^^^^^^^^^
NotADirectoryError: [Errno 20] Not a directory: '/dev/null/non-existent'

```

Reply:
interesting, is it on Mac?

Reply:
I'm on Fedora 39

Reply:
Oh, you're re-raising it, makes sense now :)